virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,117 @@
+import os
+import logging
+
+from avocado.utils import process
+from avocado.core import exceptions
+
+"""Handle Linux kernel modules"""
+
+
+class KernelModuleHandler(object):
+    """Class handling Linux kernel modules"""
+
+    def __init__(self, module_name):
+        """Create kernel module handler"""
+
+        self._module_name = module_name
+        self._was_loaded = None
+        self._loaded_config = None
+        self._config_backup = None
+        self._backup_config()
+
+    def load_module(self, force=True, params=""):
+        """
+        Load module with given parameters
+
+        :param force: Force load if currently not loaded. Default is True as
+        restore undoes loading.
+        :param params: Parameters to load with, 'key1=param1 key2=param2 ...'
+        """
+
+        current_config = KernelModuleHandler._load_config(self._module_name)
+        if current_config and all(x in current_config for x in params.split()):
+            logging.debug("Not reloading module. Current module configuration for %s"
+                          " already contains all requested parameters."
+                          " Requested: %s. Current: %s."
+                          % (self._module_name, params, current_config))
```

Comment:
Please don't use `%` formatting for logging records (5x in this module, please use `pylint` to find all issues).